### PR TITLE
Client impl

### PR DESCRIPTION
**Pull request by dorin-iancu**

### CODE REVIEW

File: common/common-modules/src/utils.rs [hunk 1]
```diff
@@ -9,4 +13,11 @@ pub trait UtilsModule {
             "Invalid address"
         );
     }
+
+    fn checked_timestamp_to_unix_mul(&self, timestamp: Timestamp) -> Timestamp {
+        match timestamp.checked_mul(NANO_SECONDS_MULT) {
```
Comment (CostinCarabas) [inline common/common-modules/src/utils.rs:18]: Here is just a value in nanoseconds rounded to seconds. If precision is needed, we need framework support. Still figuring it out if it's important

File: client-impls/local-host/src/client_logic.rs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    multiversx_sc::imports!();
+
+    #[multiversx_sc::proxy]
+    pub trait ClientProxy {
+        #[endpoint(updateClientCommitments)]
+        fn update_client_commitments(
+            &self,
+            client_id: ClientId<Self::Api>,
+            encoded_heights: ManagedBuffer,
```
Comment (CostinCarabas) [inline client-impls/local-host/src/client_logic.rs:21]: I think a `ManagedVec<height::Data>` should be here.

File: client-impls/local-host/src/client_logic.rs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    ///
+    /// `client_id`` the client identifier must be match with `CLIENT_ID`
+    #[endpoint(updateClient)]
+    fn update_client(&self, client_id: ClientId<Self::Api>) -> ManagedVec<height::Data> {
+        self.require_valid_client_id(&client_id);
+
+        let ibc_handler = self.ibc_handler().get();
+        let _: () = self
+            .client_proxy(ibc_handler)
+            .update_client_commitments(client_id, ManagedBuffer::new())
```
Comment (CostinCarabas) [inline client-impls/local-host/src/client_logic.rs:72]: TODO data about block height

**Review by CostinCarabas — APPROVED** [review]

File: client-impls/local-host/src/client_logic.rs (hunk 3 above)
Comment (dorin-iancu) [reply client-impls/local-host/src/client_logic.rs:72]: nah, this is just a test contract as far as I'm aware.

File: client-impls/local-host/src/client_logic.rs (hunk 2 above)
Comment (dorin-iancu) [reply client-impls/local-host/src/client_logic.rs:21]: Good point. Didn't realize it at the time. Will keep a list of all these things and fix them in a future PR, don't want to merge all the branches again.

File: common/common-modules/src/utils.rs (hunk 1 above)
Comment (dorin-iancu) [reply common/common-modules/src/utils.rs:18]: Doesn't seem like we will need the precision. As far as I'm aware, these timestamps are used to give a timeout for transactions.

**Pull request merged**